File: Blog.Core/Infrastructure/Pagination/Interfaces/ISortCriteria.cs
namespace Blog.Core.Infrastructure.Pagination.Interfaces
{
    using Blog.Core.Enums;
    using System.Linq;
    public interface ISortCriteria<T>
    {
        OrderType Direction { get; set; }

        IOrderedQueryable<T> ApplyOrdering(IQueryable<T> query, bool useThenBy);
    }
}

File: Shared/Blog.FSharp.Core/Infrastructure/Pagination/Interfaces/ISortCriteria.fs
namespace Blog.FSharp.Core.Infrastructure.Pagination.Interfaces

open System.Linq
open Blog.FSharp.Core.Enums

/// <summary>
/// Sort items by criteria interface.
/// </summary>
/// <typeparam name="T">Type.</typeparam>
type ISortCriteria<'T> =
    /// <summary>
    /// Gets or sets direction.
    /// </summary>
    abstract member Direction : OrderType with get, set

    /// <summary>
    /// Apply ordering.
    /// </summary>
    /// <param name="query">query.</param>
    /// <param name="useThenBy">useThenBy.</param>
    /// <returns>IOrderedQueryable.</returns>
    abstract member ApplyOrdering : query: IQueryable<'T> * useThenBy: bool -> IOrderedQueryable<'T>